do.sh

```diff
@@ -53,19 +81,37 @@ function generate() {
 
 function install_deps() {
     echo "-- Installing dependencies on all nodes"
-    ansible-playbook ${ovn_fmn_playbooks}/install-dependencies.yml -i ${hosts_file}
+    local _install_dep_playbook
+    if is_rpm_based
+    then
+        _install_dep_playbook=install-dependencies-rpm.yml
+    elif is_deb_based
+    then
+        _install_dep_playbook=install-dependencies-deb.yml
+    else
+        die_distro
+    fi
+    ansible-playbook ${ovn_fmn_playbooks}/${_install_dep_playbook} -i ${hosts_file}
 
     echo "-- Installing local dependencies"
-    if yum install docker-ce --nobest -y
+    if is_rpm_based
     then
-        systemctl start docker
+        if yum install docker-ce --nobest -y
+        then
+            systemctl start docker
+        else
+            yum install -y podman podman-docker
+        fi
+        yum install redhat-lsb-core datamash \
+            python3-pip python3-virtualenv python3 python3-devel python-virtualenv \
+            --skip-broken -y
+        [ -e /usr/bin/pip ] || ln -sf /usr/bin/pip3 /usr/bin/pip
+    elif is_deb_based
+    then
+        apt -y install datamash podman podman-docker python3-pip python3-virtualenv python3 python3-all-dev python3-virtualenv
     else
-        yum install -y podman podman-docker
+        die_distro
```

Comment:
Instead of duplicating the `die_distro` check does it make sense to just do it once, in the beginning of the `install()` function?

Reply:
In order to do this check once, we would need to change the structure of the program, i.e. we would need to install local deps first and then run the ansible playbook for other nodes afterwards, but I guess that's fine too

Reply:
Done

Reply:
Hmm, I'm sorry, I think my initial comment wasn't clear.  I actually meant this:
```
diff --git a/do.sh b/do.sh
index 6f00c08..b237600 100755
--- a/do.sh
+++ b/do.sh
@@ -303,6 +303,7 @@ function pull_ovn_tester() {
 }
 
 function install() {
+    ! is_rpm_based && ! is_deb_based && die_distro
     pushd ${rundir}
     install_deps
     install_venv
```

On a second thought, isn't it cleaner if we just add per distro `install_deps` functions.  E.g.:
```
function install() {
    pushd ${rundir}
    if is_rpm_based
    then
        install_deps_rpm
    elif is_deb_based
            install_deps_deb
    else
        die_distro
    fi
    install_venv
    configure_docker
    install_ovn_fake_multinode
    init_ovn_fake_multinode
    pull_ovn_fake_multinode
    install_ovn_tester
    pull_ovn_tester
    popd
}
```

We'll have to duplicate a bit of logic in the per-distro `install_deps_*` functions but, if we ensure that we test all supported distro types in CI, that might be acceptable.

What do you think?

Reply:
That does indeed look more appealing, makes it possible to glance at the `install` function and immediately understand what it does without all the implementation details. I'll redo this part, thx!